### PR TITLE
OpenCV 4.10.0 - Numpy 2.0

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
@conda-forge-admin, please rerender

**Comment by ilya-lavrenov** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
> Do we want to drop qt5 builds

Fine by me per se. What's the status on the overall qt6 migration? How many feedstocks (if any) are still qt5-only? I'd kinda like to drop qt5 in general...

**Comment by hmaarrfk** [comment]
Spyder is in general PyQt (not sure if they handle 6) and require webengine. Which we don't have for Qt6. 

However. All platforms have a "headless" option. So it should be co-installable, just without the GUI parts on opencv.

**Comment by h-vetinari** [comment]
Let's move this discussion to a wider forum...

**Comment by traversaro** [comment]
> What's the status on the overall qt6 migration?

I may be wrong, but I do not think any qt=5 --> qt6 migration ever started.

**Comment by h-vetinari** [comment]
The situation here is exceptional in many ways - the outputs are named differently (`qt-main` vs `qt6-main`) to support parallel installation, and so we cannot migrate as usual (until we publish a `qt-main==6.x`).

**Comment by hmaarrfk** [comment]
> The situation here is exceptional in many ways

qt is in fact, an exceptional package!

**Comment by hmaarrfk** [comment]
So the unique thing here is that users can still install opencv + spyder.

They just won't be able to get:

* Spyder + OpenCV + HighGUI for OpenCV 4.10

They owuld get

* Spyder + OpenCV Headless 4.10
or 
* Spyder + OpenCV 4.9 + HighGUI

I think that is ok.....

**Comment by jakirkham** [comment]
Worth noting that this also contains NumPy 2 support

**Comment by hmaarrfk** [comment]
ah, then maybe the right thing is to pull in the numpy 2.0 migration and "do one last qt5 build". 

I "promise it will be the last one".

**Comment by ilya-lavrenov** [comment]
@conda-forge-admin, please rerender

File: .ci_support/migrations/numpy2.yaml [hunk 1]
```diff
@@ -0,0 +1,2379 @@
+
+
+
+
+
+
+<!DOCTYPE html>
```
Comment (jakirkham) [inline .ci_support/migrations/numpy2.yaml:7]: Looks like this may not have been included correctly?

**Comment by ilya-lavrenov** [comment]
@conda-forge-admin, please rerender

**Comment by ilya-lavrenov** [comment]
Looks like numpy 2.0 is not supported by OpenCV.
They use Numpy API directly during OpenCV Python API implementation.

**Comment by traversaro** [comment]
> Looks like numpy 2.0 is not supported by OpenCV.
> They use Numpy API directly during OpenCV Python API implementation.

According to  https://github.com/opencv/opencv/issues/25455 opencv 4.10 should support numpy 2.

**Comment by hmaarrfk** [comment]
There is probably something off in our windows installation script.

**Comment by hmaarrfk** [comment]
Back when I started to get involved in CF, one of my major contributions was a refactor to this installation script.
https://github.com/ilya-lavrenov/opencv-feedstock/blob/opencv-4.10.0/recipe/bld.bat#L117

there is probably alot left over from the python 2.7/3 days that can be removed.

If you do end up attempting a cleanup Ilya, please skip most of the jobs by adding something like:

```
skip: true   # [py!=310]
skip: true   # [ffmpeg == 6]
```
and troubleshoot with that.

**Pull request merged**